content/news/posts/2024/meet-prestashop9-api.md

```diff
@@ -0,0 +1,386 @@
+
+* **client_id** - the ID of the client you created
+* **client_secret** - the secret of the client you created
+* **grant_type** - the type of grant you want to use. In this case, it's `client_credentials`
+* **scope** - the scopes you want to use. You can provide multiple scopes. In this case, we are going to use `api_client_read`, `api_client_write`, `customer_group_read`, `customer_group_write`.
+
+Example request using curl:
+
+```bash
+curl --location 'http://yourdomain.test/admin-api/access_token' \
+--form 'grant_type="client_credentials"' \
+--form 'client_id="your_client_id"' \
+--form 'client_secret="your_client_secret"' \
+--form 'scope[]="api_client_read"' \
+--form 'scope[]="api_client_write"' \
+--form 'scope[]="customer_group_read"' \
+--form 'scope[]="customer_group_write"'
+```
+
+Example request using Postman:
+
+![PrestaShop 9 API Access Token request using Postman](/assets/images/2024/05/api/postman_access_token.png)
+<p class="text-center text-muted small">Screenshot from Postman where we retrieve access token for our client</p>
+
+{{% notice %}}
+As you can see in the screenshot, I am using project variables in Postman. This way, you can easily switch between different environments (like development, staging, and production) and don't have to change the request data (URL, client_id, etc.) every time you switch environments.
+{{% /notice %}}
+
+After you send the request, you should receive a response with the access token. If you receive information saying "Use HTTPS response", it means that you need to use HTTPS to connect to the API. As mentioned earlier, you can turn off this check in the PrestaShop back office.
+
+If you still have problems with the connection, make sure that the API is enabled in the PrestaShop back office. You can check this in the Advanced Parameters -> Admin API -> Configuration page.
+
+If everything goes well, the response should look similar to this:
+
+```json
+{
+    "token_type": "Bearer",
+    "expires_in": 3600,
+    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiJ0ZXN0X2NsaWVudCIsImp0aSI6ImQ5OTQzYWQzNDgwNDA3N2QyZGQ5MTBmM2E3YTVmYTdiZTQ1YmMyNzZmM2VmMjA2MGJmZjg2MzhlMzA4ZWE3OWY1ZWI1NzQ0ZmI0N2Y4MDU2IiwiaWF0IjoxNzE2Mjg2MjE5LjQ2NDA3OCwibmJmIjoxNzE2Mjg2MjE5LjQ2NDA4LCJleHAiOjE3MTYyODk4MTkuNDYzMDEsInN1YiI6IiIsInNjb3BlcyI6WyJpc19hdXRoZW50aWNhdGVkIiwiYXBpX2NsaWVudF9yZWFkIiwiYXBpX2NsaWVudF93cml0ZSIsImN1c3RvbWVyX2dyb3VwX3JlYWQiLCJjdXN0b21lcl9ncm91cF93cml0ZSJdfQ.Q6kK0Pl1HVAVrzn5xUrzRO1VSUaw-ygTn9D_rKlfjW3gllUWJiWRaA_pM53RtLId1LkcAfW8nW27CFhQH7TQqLCn4vUPD2t6_s7-3WX_HIqe6MHExib2mW7u_ZXT3bSOyPUOjWIcZsISQR1-noZfOEYDkvgnKDC250zieVqMELgxclMFXKdiLhn83GJnCW35llB1TwAACxdV1uJ_emZGCR3Tsy2IK1pSKPRAb2h-gBre8hqtCmUZ5pdM_L6D_EuUM-aB6iQENiCD6ECmSvqvsqkd3RB73s7PntwniUUafD2GHap1Ttw8pOF7omtT3X0ZLssSX1eQMPDw6JGLFz9caw"
```

Comment:
I don't know if we should indicate that the token can be parsed on jwt.io to see its content? Is it too technical for a first user guide?